pixel_clusterizer/cluster_functions.py

```diff
@@ -1,182 +1,277 @@
+        hits=hits,
+        cluster=cluster,
+        cluster_size=cluster_size,
+        cluster_hit_indices=cluster_hit_indices,
+        cluster_index=cluster_index,
+        cluster_id=cluster_id,
+        charge_correction=charge_correction,
+        noisy_pixels=noisy_pixels,
+        disabled_pixels=disabled_pixels,
+        seed_hit_index=seed_hit_index)
+
+
+@njit()
+def _finish_event(hits, cluster, start_event_hit_index, stop_event_hit_index, start_event_cluster_index, stop_event_cluster_index):
+    ''' Set hit and cluster information of the event (e.g. number of cluster in the event (n_cluster), ...).
+    '''
+    for hit_index in range(start_event_hit_index, stop_event_hit_index):
+        hits[hit_index]['n_cluster'] = stop_event_cluster_index - start_event_cluster_index
+
+    for cluster_index in range(start_event_cluster_index, stop_event_cluster_index):
+        cluster[cluster_index]['event_number'] = hits[start_event_hit_index]['event_number']
 
     # Call end of event function hook
-    _end_of_event_function(hits, cluster, is_seed, n_cluster, cluster_size, cluster_id, actual_event_hit_index, new_actual_event_hit_index, next_cluster_id, actual_event_cluster_index)
+    _end_of_event_function(
+        hits=hits,
+        cluster=cluster,
+        start_event_hit_index=start_event_hit_index,
+        stop_event_hit_index=stop_event_hit_index,
+        start_event_cluster_index=start_event_cluster_index,
+        stop_event_cluster_index=stop_event_cluster_index)
+
+
+@njit()
+def _hit_ok(hit, min_hit_charge, max_hit_charge):
+    ''' Check if given hit is withing the limits.
+    '''
+    # Omit hits with charge < min_hit_charge
+    if hit['charge'] < min_hit_charge:
+        return False
+
+    # Omit hits with charge > max_hit_charge
+    if max_hit_charge != 0 and hit['charge'] > max_hit_charge:
+        return False
+
+    return True
+
+
+@njit()
+def _set_hit_invalid(hit, cluster_id=-1):
+    ''' Set values for invalid hit.
+    '''
+    hit['cluster_ID'] = cluster_id
+    hit['is_seed'] = 0
+    hit['cluster_size'] = 0
 
 
 @njit()
-def _reset_cluster_hit_indices(actual_cluster_hit_indices, actual_cluster_size):
-    ''' Sets the cluster hit indices to the std. valie -1. To be able to use this array in the new event'''
-    for i in range(actual_cluster_size):
-        actual_cluster_hit_indices[i] = -1
+def _set_1d_array(array, value, size=-1):
+    ''' Set array elemets to value for given number of elements (if size is negative number set all elements to value).
+    '''
+    if size >= 0:
+        for i in range(size):
+            array[i] = value
+    else:
+        for i in range(array.shape[0]):
+            array[i] = value
 
 
 @njit()
 def _is_in_max_difference(value_1, value_2, max_difference):
     ''' Helper function to determine the difference of two values that can be np.uints. Works in python and numba mode.
-    Circumvents numba bug #1653'''
+    Circumvents numba bug #1653
+    '''
```

Comment:
Yes. Generally there is a lot of room for better code formatting this package ;-)